src/internal.c

```diff
@@ -7755,14 +7839,19 @@ int SendKexInit(WOLFSSH* ssh)
     word32 idx = 0;
     word32 payloadSz = 0;
     int ret = WS_SUCCESS;
-    const char* cannedKeyAlgoNames = NULL;
-    word32 cannedKeyAlgoNamesSz = 0;
+    word32 keyAlgoNamesSz = 0;
+    char* keyAlgoNames = NULL;
 
     WLOG(WS_LOG_DEBUG, "Entering SendKexInit()");
 
     if (ssh == NULL)
         ret = WS_BAD_ARGUMENT;
 
+    if (ssh->ctx->side == WOLFSSH_ENDPOINT_SERVER &&
+            ssh->ctx->privateKeyCount == 0) {
+        ret = WS_BAD_ARGUMENT;
```

Comment:
Could be a good place for a debug message saying no private key loaded.

Reply:
check